commit 318ff841ff1dee3a2572b9efaab243aed896b64e
Author: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Date:   Fri Jan 22 16:48:46 2016 +0200

    mfd: intel-lpss: Remove clock tree on error path
    
    commit 84cb36cac581c915ef4e8b70abb73e084325df92 upstream.
    
    We forgot to remove the clock tree if something goes wrong in ->probe(). Add a
    call to intel_lpss_unregister_clock() on error path in ->probe() to fix the
    potential issue.
    
    Fixes: 4b45efe85263 (mfd: Add support for Intel Sunrisepoint LPSS devices)
    Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
    Acked-by: Mika Westerberg <mika.westerberg@linux.intel.com>
    Signed-off-by: Lee Jones <lee.jones@linaro.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/mfd/intel-lpss.c b/drivers/mfd/intel-lpss.c
index 1743788..1bbbe87 100644
--- a/drivers/mfd/intel-lpss.c
+++ b/drivers/mfd/intel-lpss.c
@@ -453,6 +453,7 @@ int intel_lpss_probe(struct device *dev,
 err_remove_ltr:
 	intel_lpss_debugfs_remove(lpss);
 	intel_lpss_ltr_hide(lpss);
+	intel_lpss_unregister_clock(lpss);
 
 err_clk_register:
 	ida_simple_remove(&intel_lpss_devid_ida, lpss->devid);
